### DISCUSSION TITLE
dumpdata and PickledObjectField

### DISCUSSION THREAD

**Discussion by dvoraen** [evennia/evennia#2805] [Evennia Questions] [answered]
Hello!  So when I'm trying to use dumpdata through the Evennia launcher, I get the following back:

```python
CommandError: Unable to serialize database: 'PickledObjectField' object has no attribute '_get_val_from_obj'
Exception ignored in: <generator object cursor_iter at 0x0000020B624383C0>
Traceback (most recent call last):
  File "F:\ArxDev\arxcode\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1557, in cursor_iter
    cursor.close()
sqlite3.ProgrammingError: Cannot operate on a closed database.
```
As far as I can tell, there is indeed no such method named `_get_val_from_obj()` in `PickledObjectField` itself, or inherited via django's `models.Field`.  I can't seem to find any calls made to `PickledObjectField.value_to_string()` in Evennia either, which is the ONLY use of `_get_val_from_obj()` that I can see, so I'm a little baffled as to its purpose.  Is there something I'm missing, or is this basically defunct code?  I'd like to be able to flop my database into the JSON at will so I can better understand how the data is stored, but this error is blocking me.

Thank you for your time.

**Comment by Griatch** [comment, marked as answer]
`PickledObjectField` is a custom field type, but it should work fine with `dump`; if it does not, this looks like it belongs in a [bug ticket](https://github.com/evennia/evennia/issues/new/choose). 

As for investigating and understanding your database, may I suggest inspecting it using a sqlite browser? `sqlitebrowser` is good, otherwise there are even browser-based ones you can use.

**Reply by dvoraen** [reply]
Noted on sqlitebrowser.  That will be a nice tool, thanks for alerting me to it.  I'm going to tinker with dumpdata some more since it's erroring back at me for different reasons on different apps so there may be something on my end that's causing a problem.
